rust/src/applayer.rs

```diff
@@ -116,6 +116,10 @@ pub struct AppLayerTxData {
     pub file_tx: u8,
     /// Number of times this tx data has already been logged for one stream match
     pub stream_logged: u8,
+    /// The tx has been processed : detection, logging, cleaning
+    /// It can be skipped until new data arrives.
+    /// There is a boolean for both directions : to server and to client
+    pub processed_until_update: [bool; 2],
```

Comment:
I feel it would make more sense to flip the logic here: call this `updated` and set it to true if the parser updated it, then reset to false when processing of detect/log/etc has been done.

Reply:
this would get rid of the `false` sets that seem a bit awkward to me, and would replace them with a more logical `true` set on modification

Reply:
ok